tests/unit/composition/Composition.test.ts

```
@@ -0,0 +1,147 @@
+      (sum: number, val: number) => sum + (val || 0),
+      0,
+    ),
+  ),
+  composition_to_percentages: vi.fn((comp: Record<string, number>) => {
+    const total = Object.values(comp).reduce(
+      (sum: number, val: number) => sum + (val || 0),
+      0,
+    )
+    const percentages: Record<string, number> = {}
+    for (const [element, amount] of Object.entries(comp)) {
+      if (typeof amount === `number`) {
```

Biome (1.9.4)

[error] Invalid `typeof` comparison value: this expression is not a string literal

not a string literal

(lint/suspicious/useValidTypeof)